src/viam/sdk/log/logger.cpp

```diff
@@ -0,0 +1,160 @@
+    } else if (level == "warn" || level == "warning") {
+        return ll::warn;
+    } else if (level == "error") {
+        return ll::error;
+    } else if (level == "debug") {
+        return ll::debug;
+    } else if (level == "trace") {
+        return ll::trace;
+    } else if (level == "fatal") {
+        return ll::fatal;
+    }
+
+    VIAM_LOG(warn) << "Returning unknown log level `" << level << "` as info";
+    return ll::info;
+}
+
+std::ostream& operator<<(std::ostream& os, log_level lvl) {
+    os << to_string(lvl);
+    return os;
+}
+
+std::string global_resource_name() {
+    return "Viam C++ SDK";
+}
+
+bool Logger::Filter::operator()(const boost::log::attribute_value_set& attrs) const {
+    auto sev = attrs[attr_sev_type{}];
+    if (!sev) {
+        return false;
+    }
+
+    auto resource = attrs[attr_channel_type{}];
+    if (resource) {
+        auto it = parent->resource_levels_.find(*resource);
+        if (it != parent->resource_levels_.end()) {
+            return *sev >= it->second;
+        }
+    }
+
+    return *sev >= parent->global_level_;
+}
+
+Logger& Logger::get() {
+    static Logger& result = Instance::current(Instance::Creation::open_existing).impl_->logger;
+
+    return result;
+}
+
+LogSource& Logger::logger() {
+    return sdk_logger_;
+}
+
+void Logger::set_global_log_level(log_level lvl) {
+    global_level_ = lvl;
+}
+
+void Logger::set_global_log_level(int argc, char** argv) {
+    if (argc >= 3 && strcmp(argv[2], "--log-level=debug") == 0) {
+        set_global_log_level(log_level::debug);
+    }
+}
+
+void Logger::set_resource_log_level(const std::string& resource, log_level lvl) {
+    resource_levels_[resource] = lvl;
+}
+
+void Logger::init_logging() {
+    sdk_logger_.channel(global_resource_name());
+    boost::log::core::get()->add_global_attribute("TimeStamp",
+                                                  boost::log::attributes::local_clock());
+
+    boost::log::formatter fmt =
+        boost::log::expressions::stream
+        << boost::log::expressions::format_date_time<boost::posix_time::ptime>(
+               "TimeStamp", "%Y--%m--%d %H:%M:%S")
+        << ": [" << attr_channel_type{} << "] <" << attr_sev_type{} << "> [" << attr_file_type{}
+        << ":" << attr_line_type{} << "] " << boost::log::expressions::smessage;
+
+    auto backend = boost::make_shared<boost::log::sinks::text_ostream_backend>();
+    backend->add_stream(boost::shared_ptr<std::ostream>(&std::cout, boost::null_deleter()));
+    backend->auto_flush(true);
+
+    console_sink_ = boost::make_shared<
+        boost::log::sinks::synchronous_sink<boost::log::sinks::text_ostream_backend>>(backend);
+
+    console_sink_->set_filter(Filter{this});
+    console_sink_->set_formatter(fmt);
+
+    boost::log::core::get()->add_sink(console_sink_);
+    VIAM_LOG(info) << "Initialized console logging";
+}
+
+void Logger::disable_console_logging() {
+    VIAM_LOG(info) << "Disabling console logging";
```

Comment:
(nit, opt) I'm probably over-thinking things but will modules always start up with a `Initialized console logging` log message immediately followed by a `Disabling console logging` message? That seems a little noisy. Maybe better to relegate these to `debug` level?